docs/proposals/pepxxx_build_isolation_passthrough.md

```diff
@@ -0,0 +1,123 @@
+| `PEP Link`          | `To Be Published`                                                   |
+| `DPO Discussion`    | [\[Build Isolation Passthrough\] Allowing a "package” to passthrough the build isolation](<https://discuss.python.org/t/build-isolation-passthrough-allowing-a-specific-package-to-always-passthrough-the-build-isolation/77284/>) |
+| `Github Repository` |                                                                     |
+
+## Abstract
+
+In modern software development, build isolation is a critical practice that ensures the consistency and reliability of
+builds. It involves creating an isolated environment where dependencies and build tools are controlled and do not
+interfere with the system or other projects. However, there are scenarios where certain tools or dependencies need to
+be accessed directly from the host environment.
+
+This proposal outlines a strategy to enhance the build isolation mechanism introduced by
+[PEP 518](http://peps.python.org/pep-0518/). Build isolation passthrough is a mechanism that allows
+selective access to host resources while maintaining the integrity of the isolated environment.
+This feature aims to address the challenges faced when a package depends on the ABI of another package, particularly in
+scenarios involving C/C++/Rust/CUDA/etc. dependencies.
+
+## Motivation
+
+With the adoption of `pyproject.toml` as specified in [PEP 518](http://peps.python.org/pep-0518/), build isolation has
+become a standard practice in Python packaging. However, there are cases where build isolation can be counterproductive,
+especially when dealing with packages that depend on the ABI of other packages. This proposal aims to provide a solution
+for such cases, improving the efficiency and flexibility of the build process.
+
+## Rationale
+
+### Current Challenges
+
+1. **Version Mismatch**: During the build phase, pip may download a different version of a dependency than the one
+present on the user's machine, leading to potential compatibility issues.
+2. **Unreleased/Local Dependencies**: Some dependencies may not be publicly released or may exist only as development versions
+on a developer's machine.
+3. **Large Dependencies**: Downloading large dependencies repeatedly can significantly increase build times, especially
+for packages with large binary dependencies like Artificial Intelligence libraries and frameworks.
+
+### Proposed Solution
+
+Allowing specific packages to opt-in to bypass build isolation for a one or many packages already present on the user's
+machine can mitigate these issues. This approach would reduce unnecessary downloads, ensure compatibility with the
+user's environment, and decrease build times.
+
+## Specification
+
+### pyproject.toml Configuration
+
+The proposed feature would allow specifying packages that should bypass build isolation in the `pyproject.toml` file.
+
+For example:
+
+```toml
+[tool.uv]
```

Comment:
This shouldn't be tool-specific, both because then it wouldn't require a PEP and because it's ineffective - it should work for any build frontend.

Naming is hard and all, and there's a question of whether this should be a new table or under `[project]`, but for now I'd suggest to use `[project]`.

Reply:
I totally agree ! But this is a good example of a "working implementation"

Reply:
True. `[tool.uv]` is also further down though, under Reference Implementation, which is the right place I think. Under Specification it doesn't really fit.

Reply:
Ah right I'll change that